### PR TITLE
 Add warning if observed in DensityDist is dict

### PR DESCRIPTION
**Pull request by symeneses**
**What is this PR about?**
It adds a warning if the observed parameter is a dictionary.

**Checklist**
+ [ x] Explain important implementation details 👆
+ [ x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x ] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x ] Are the changes covered by tests and docstrings?
+ [ x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...
It solves #6032 
## Docs / Maintenance
- ...

### CODE REVIEW

**Comment by fonnesbeck** [comment]
LGTM. Thanks for contributing!

**Review by fonnesbeck — APPROVED** [review]

File: pymc/distributions/distribution.py [hunk 1]
```diff
@@ -577,6 +577,10 @@ def random(mu, rng=None, size=None):
 
     def __new__(cls, name, *args, **kwargs):
         kwargs.setdefault("class_name", name)
+        if "observed" in kwargs.keys() and isinstance(kwargs["observed"], dict):
```
Comment (michaelosthege) [inline pymc/distributions/distribution.py]:
```suggestion
        if isinstance(kwargs.get("observed", None), dict):
```

File: pymc/distributions/distribution.py [hunk 2]
```diff
@@ -577,6 +577,10 @@ def random(mu, rng=None, size=None):
...
+            raise TypeError(
+                "The observed parameter should be of type `pd.Series`, " "`np.array`, or `pm.Data`."
```
Comment (michaelosthege) [inline pymc/distributions/distribution.py]:
```suggestion
                "The observed parameter should be of type `pd.Series`, `np.array`, or `pm.Data`."
```

**Review by michaelosthege — APPROVED** [review]
just some code style simplifications/FYI

looks good!

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
This is a great start but misses the most important point of the original issue. 

DensityDist is defined differently than how it was in V3 where a dictionary with the other parameters besides data could be used. The error message should address this change in API more than what types of observed data are allowed (that's not what users find confusing).

**Comment by symeneses** [comment]
Based on this item in the Release notes:
> pm.DensityDist now accepts distribution parameters as positional arguments. Passing them as a dictionary in the observed keyword argument is no longer supported and will raise an error (see 5026).

I am suggesting this new error message:

Since ``v4.0.0`` the ``observed`` parameter should be of type ``pd.Series``, ``np.array``, or ``pm.Data``.
Previous versions allowed passing distribution parameters as a dictionary in ``observed``, in the current version these parameters are positional arguments.

Let me know if this message can be further improved.

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**